### PR TITLE
Vfx/docs/bulk review

### PR DESCRIPTION
**Pull request by JordanL8**
### Purpose of this PR
Reviewed Exposed Properties and Component API docs.

### CODE REVIEW

**Comment by julienf-unity** [comment]
@PaulDemeulenaere Can you check and confirm everything is correct in the doc regarding effect life cycle, exposed properties and event attributes?

**Comment by PaulDemeulenaere** [comment]
> @PaulDemeulenaere Can you check and confirm everything is correct in the doc regarding effect life cycle, exposed properties and event attributes?

I'm checking this today (I have just received mail notification, sorry for the delay).

File: com.unity.visualeffectgraph/Documentation~/ComponentAPI.md [hunk 1]
```diff
@@ -1,137 +1,179 @@
...
+* In the component API : `initialEventName = "MyEventName";`.
+* In the component API : `initialEventID = Shader.PropertyToID("MyEventName";`.
+* Using the [ExposedProperty Helper Class](ExposedPropertyHelper.md).
+
+## Random seed control
+
+Every effect instance has settings and controls for its random seed. You can modify the seed to influence the random values the Visual Effect Graph uses.
 
-You can override this property:
+* `resetSeedOnPlay = true/false`: Controls whether Unity computes a new random seed every time you call the `Play()` function. This causes each random value the Visual Effect Graph uses to be different to what it was in previous simulations.
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Well, actually, resetSeedOnPlay is wrongly named and make reference to the enter state "Play" corresponding to an enabled state or change to PlayMode.
So, it's  "you call the `Play()`" =>  "you call the `ReInit()`" (this error was already there, I missed it)
Maybe there is still this inconsistency in C++ documentation. We should renamed this property.

File: com.unity.visualeffectgraph/Documentation~/ComponentAPI.md [hunk 2]
```diff
@@ -1,137 +1,179 @@
...
+* `SetMatrix4x4(property,value)`
 
-Every Set method will set the corresponding value to the property and will set it as overridden.
+Each method overrides the value of the corresponding property with the value you pass in.
 
-#### Resetting Property Overrides and Default Values
+### Resetting property overrides and default values
 
-Resetting Property Overrides on a  [Visual Effect Component](VisualEffectComponent.md) can be achieved using the `ResetOverride(property)`.
+The component API allows you to reset property overrides back to their original values. To do this, use the `ResetOverride(property)` method.
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Note for the future, there is an issue with serialization using ResetOverride from script : https://fogbugz.unity3d.com/f/cases/1206890/

File: com.unity.visualeffectgraph/Documentation~/ComponentAPI.md [hunk 3]
```diff
@@ -1,137 +1,179 @@
...
+Event Attributes are [Attributes](Attributes.md) that attach to [Events](Events.md) and can be processed by the Visual Effect Graph. To create and store Event Attributes, use the `VFXEventAttribute` class. The Visual Effect component is responsible for creating instances of the `VFXEventAttribute` class and creates them based on the currently assigned [Visual Effect Graph](VisualEffectGraphAsset.md).
 
 #### Creating Event Attributes
 
-In order to Create and Use a `VFXEventAttribute` use the `CreateVFXEventAttribute()` method of the `VisualEffect` component. If you plan on sending multiple times events using attributes, you will preferably cache this object so you can reuse it.
+To create a `VFXEventAttribute`, use the `CreateVFXEventAttribute()` method of the Visual Effect component. If you want to send the same Event multiple times with the same attributes, you should store the `VFXEventAtrribute`, rather than create a new one every time you send the Event.
```
Comment (PaulDemeulenaere) [inline com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Maybe, we can mention an implicit behavior :
You can modify an EventAttribute after a SendEvent, the state while calling VisualEffect.SendEvent has been already copied.

Comment (JordanL8) [reply com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Something like:

When you send an Event to a Visual Effect Graph, Unity creates a copy of the EventAttribute in its current state and sends the copy. This means that, after you send the Event, you can safely modify the EventAttribute without affecting the information sent to the Visual Effect Graph.

File: com.unity.visualeffectgraph/Documentation~/ComponentAPI.md [hunk 4]
```diff
@@ -1,137 +1,179 @@
...
 ### Default Event
 
-Upon enabling the GameObject, or the Component, a default event is sent to the graph : by default it is `OnPlay` which is the implicit Start of [Spawn Contexts](Contexts.md#spawn).
+When the Visual Effect component (or the GameObject it attaches to) enables, it sends an [Event](Events.md) to the graph. By default, this Event is `OnPlay` which is the standard start for [Spawn Contexts](Contexts.md#spawn).
+
+You can change the default Event in the following ways:
+
+* On the [Visual Effect Inspector](VisualEffectComponent.md), change the **Initial Event Name** field.
+* In the component API : `initialEventName = "MyEventName";`.
+* In the component API : `initialEventID = Shader.PropertyToID("MyEventName";`.
```
[the remark this answers is not in the document]
Comment (JordanL8) [reply com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Fixed in https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/5907/commits/6b59e0757231ffac8fb840b4cad34f8bef36b464

File: com.unity.visualeffectgraph/Documentation~/ComponentAPI.md (hunk 1 above)
Comment (JordanL8) [reply com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Fixed in https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/5907/commits/6b59e0757231ffac8fb840b4cad34f8bef36b464

File: com.unity.visualeffectgraph/Documentation~/ComponentAPI.md (hunk 3 above)
Comment (JordanL8) [reply com.unity.visualeffectgraph/Documentation~/ComponentAPI.md]: Added suggestion in https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/5907/commits/6b59e0757231ffac8fb840b4cad34f8bef36b464

**Pull request merged**